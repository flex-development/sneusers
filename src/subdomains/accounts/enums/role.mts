/**
 * @file Enums - Role
 * @module sneusers/accounts/enums/Role
 */

/**
 * User roles.
 *
 * @enum {Lowercase<string>}
 */
enum Role {
  DEVELOPER = 'developer',
  USER = 'user'
}

export default Role
