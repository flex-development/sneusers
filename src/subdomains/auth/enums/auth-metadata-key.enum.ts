/**
 * @file Auth Subdomain Enums - AuthMetadataKey
 * @module sneusers/subdomains/auth/enums/AuthMetadataKey
 */

/**
 * Custom metadata keys.
 *
 * @see https://docs.nestjs.com/fundamentals/execution-context#reflection-and-metadata
 *
 * @enum {Uppercase<string>}
 */
enum AuthMetadataKey {
  JWT_OPTIONAL = 'JWT_OPTIONAL'
}

export default AuthMetadataKey
