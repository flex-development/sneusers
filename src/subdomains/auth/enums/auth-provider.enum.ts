/**
 * @file Auth Subdomain Enums - AuthProvider
 * @module sneusers/subdomains/auth/enums/AuthProvider
 */

/**
 * Authentication providers.
 *
 * @enum {Lowercase<string>}
 */
enum AuthProvider {
  GITHUB = 'github'
}

export default AuthProvider
