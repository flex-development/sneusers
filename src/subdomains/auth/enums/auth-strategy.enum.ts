/**
 * @file Auth Subdomain Enums - AuthStrategy
 * @module sneusers/subdomains/auth/enums/AuthStrategy
 */

/**
 * Types of authentication strategies.
 *
 * @enum {Lowercase<string>}
 */
enum AuthStrategy {
  GITHUB = 'github',
  GOOGLE = 'google',
  JWT = 'jwt-auth',
  JWT_REFRESH = 'jwt-refresh',
  LOCAL = 'local'
}

export default AuthStrategy
