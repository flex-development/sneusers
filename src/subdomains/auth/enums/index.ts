/**
 * @file Auth Subdomain Enums - Entry Point
 * @module sneusers/subdomains/auth/enums
 */

export { default as TokenType } from './token-type.enum'
export { default as VerifType } from './verif-type.enum'
