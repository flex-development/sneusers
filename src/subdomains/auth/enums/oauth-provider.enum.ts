/**
 * @file Auth Subdomain Enums - OAuthProvider
 * @module sneusers/subdomains/auth/enums/OAuthProvider
 */

/**
 * OAuth providers.
 *
 * @enum {Lowercase<string>}
 */
enum OAuthProvider {
  GITHUB = 'github',
  GOOGLE = 'google'
}

export default OAuthProvider
