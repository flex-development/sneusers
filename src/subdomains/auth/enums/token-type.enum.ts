/**
 * @file Auth Subdomain Enums - TokenType
 * @module sneusers/subdomains/auth/enums/TokenType
 */

/**
 * Types of JWTs issued.
 *
 * @enum {Uppercase<string>}
 */
enum TokenType {
  ACCESS = 'ACCESS',
  REFRESH = 'REFRESH',
  VERIFICATION = 'VERIFICATION'
}

export default TokenType
