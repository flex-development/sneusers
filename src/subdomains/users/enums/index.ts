/**
 * @file Users Subdomain Entry Point - Enums
 * @module sneusers/subdomains/users/enums
 */

export {}
