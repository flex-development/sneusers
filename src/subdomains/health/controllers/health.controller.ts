/**
 * @file Controllers - HealthController
 * @module sneusers/subdomains/health/controllers/Health
 */

import { Endpoint, Subdomain } from '#src/enums'
import { Controller, Get, HttpStatus, Inject } from '@nestjs/common'
import {
  ApiOkResponse,
  ApiServiceUnavailableResponse,
  ApiTags
} from '@nestjs/swagger'
import {
  HealthCheckService,
  HttpHealthIndicator,
  MikroOrmHealthIndicator,
  type HealthCheckResult
} from '@nestjs/terminus'
import { getHealthCheckSchema } from '@nestjs/terminus/dist/health-check/health-check.schema'

/**
 * Health check controller.
 *
 * @see https://docs.nestjs.com/recipes/terminus
 *
 * @class
 */
@Controller(Endpoint.HEALTH)
@ApiTags(Subdomain.HEALTH)
class HealthController {
  /**
   * Creates a new health check controller.
   *
   * @param {HealthCheckService} health - Health check service
   * @param {HttpHealthIndicator} http - HTTP health indicator
   * @param {MikroOrmHealthIndicator} mikro - MikroORM health indicator
   */
  constructor(
    @Inject(HealthCheckService)
    private readonly health: HealthCheckService,
    @Inject(HttpHealthIndicator)
    private readonly http: HttpHealthIndicator,
    @Inject(MikroOrmHealthIndicator)
    private readonly mikro: MikroOrmHealthIndicator
  ) {}

  /**
   * Health check generator.
   *
   * @public
   * @async
   *
   * @return {Promise<HealthCheckResult>} Health check result
   */
  @Get()
  @ApiOkResponse({
    description: 'Health check successful',
    schema: getHealthCheckSchema('ok'),
    status: HttpStatus.OK
  })
  @ApiServiceUnavailableResponse({
    description: 'Health check failed',
    schema: getHealthCheckSchema('error'),
    status: HttpStatus.SERVICE_UNAVAILABLE
  })
  public async get(): Promise<HealthCheckResult> {
    return this.health.check([
      async () => this.mikro.pingCheck(Subdomain.DATABASE),
      async () => this.http.pingCheck(Subdomain.DOCS, Endpoint.DOCS)
    ])
  }
}

export default HealthController
