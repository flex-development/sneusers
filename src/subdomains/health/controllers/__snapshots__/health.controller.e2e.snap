// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`e2e:subdomains/health/controllers/HealthController > /health > GET > should respond with health check result 1`] = `
{
  "details": {
    "docs": {
      "status": "up",
    },
  },
  "error": {},
  "info": {
    "docs": {
      "status": "up",
    },
  },
  "status": "ok",
}
`;
